charts/telemetry-collector/templates/configmap.yaml

```
@@ -0,0 +1,78 @@
+{{- include
+      "accelleran.common.configMap"
+      (fromYaml (include "accelleran.telemetry-collector.configMap.args" .))
+-}}
+
+{{- define "accelleran.telemetry-collector.configMap.args" -}}
+{{- $ := . -}}
+
+top:
+  {{ $ | toYaml | nindent 2 }}
+
+name: {{ include "accelleran.common.fullname" (dict "top" $) }}-config
+
+data:
+  config.yaml: |-
+    internal:
+      kafka:
+        url: "{{ include "accelleran.common.bootstrap.kafka.hostname" (dict "top" .) }}:{{ include "accelleran.common.bootstrap.kafka.port" (dict "top" .) }}"
+      {{- if $.Values.config.influxDB }}
+      influxDB:
+        {{- if $.Values.config.influxDB.url }}
+        url: "{{ $.Values.config.influxDB.url }}"
+        {{ else }}
+        url: "http://{{ $.Release.Name }}-influxdb2 }}"
+        {{- end }}
+        org: "{{ $.Values.config.influxDB.org }}"
+      {{- end }}
+    pipelines:
+      {{- $natsHostname := include "accelleran.common.bootstrap.nats.hostname" (dict "top" .) }}
+      {{- $natsPort := include "accelleran.common.bootstrap.nats.port" (dict "top" .) }}
+      {{- range $pipeline := $.Values.config.pipelines }}
+      - name: "{{ $pipeline.name }}" 
+        component: "{{ $pipeline.component }}"
+        vendor: "{{ $pipeline.vendor }}"
+        enabled: "{{ $pipeline.enabled }}"
+        {{- if $pipeline.consumers }}
+        consumers:
+          {{- if $pipeline.consumers.natsConsumer }}
+            natsConsumer:
+              {{- if $pipeline.consumers.natsConsumer.url }}
+              url: "{{ $pipeline.consumers.natsConsumer.url }}"
+              {{ else }}
+              url: "{{ $natsHostname }}:{{ $natsPort }}"
+              {{- end }}
+              subjects: "{{ $pipeline.consumers.natsConsumer.subjects }}"
+          {{- end }}
+          {{- if $pipeline.consumers.udpConsumer }}
+            udpConsumer:
+              port: "{{ $pipeline.consumers.udpConsumer.port }}"
+          {{- end }}
+        {{- end }}
+        {{- if $pipeline.publishers }}
+        publishers:
+          {{- if $pipeline.publishers.kafkaPublisher }}
+            kafkaPublisher:
+              topic: "{{ $pipeline.publishers.kafkaPublisher.topic }}"
+          {{- end }}
+          {{- if $pipeline.publishers.influxPublisher }}
+            influxPublisher:
+              {{- if $pipeline.publishers.influxPublisher.defaultBucket }}
+              defaultBucket: "{{ $pipeline.publishers.influxPublisher.defaultBucket }}"
+              {{- end }}
+              {{- if $pipeline.publishers.influxPublisher.buckets }}
+              buckets: {
+                {{- range $key, $value := $pipeline.publishers.influxPublisher.buckets }}
+                  "{{ $key }}": "{{ $value }}",
+                {{- end }}
+              }
+              {{- end }}
+          {{- end }}
+        {{- end }}
+        vendorConfig: 
+        {{- range $key, $value := $pipeline.vendorConfig }}
+          {{ $key }}: "{{ $value }}"
+        {{- end }}
```

I would just use the `toYaml` `(n)indent` function comba as it basically does this for us, plus it allows to have more nesting in later additions:

```suggestion
        vendorConfig: {{ $pipeline.vendorConfig | toYaml | nindent 10 }}
```

I also don't know if we maybe want to include `with` around it? (The difference is having it explicitly there set as `null`, or leaving it out)

---

We don't now have any vendor Configs - but I have added this using a with